Core/contentblocker.js

```diff
@@ -553,29 +553,39 @@ _utf8_encode : function (string) {
     // Init
     (function() {
         
-        duckduckgoDebugMessaging.log("installing beforeload detection")
-        document.addEventListener("beforeload", function(event) {
-
-            if (event.target.nodeName == "LINK") {
-                type = event.target.rel
-            } else if (event.target.nodeName == "IMG") {
-                type = "image"
-            } else if (event.target.nodeName == "IFRAME") {
-                type = "subdocument"
-            } else {
-                type = event.target.nodeName
-            }
-
-            duckduckgoDebugMessaging.log("checking " + event.url + " (" + type + ")");
-            if (shouldBlock(event.url, type)) {
-                duckduckgoDebugMessaging.log("blocking beforeload")
-                event.preventDefault()
-                event.stopPropagation()
-            } else {
-                duckduckgoDebugMessaging.log("don't block " + event.url);
-                return
-            }
-        }, true)
+        duckduckgoDebugMessaging.log("installing load detection")
+        window.addEventListener("load", function(event) {
```

Comment:
this means that we will get one snapshot and won't see other stuff loaded dynamically after page has loaded, right?

Maybe we should should check for changes every X instead?

Reply:
I think this is ok because there are other hooks to check for other content being loaded.

Reply:
Put this on an interval of x * 2 with x = 2 to start per conversation with @bwaresiak